src/AMQPEndpoint.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+/**
+ * Base class for AMQPRPCServer/AMQPRPCClient.
+ *
+ * @class
+ */
+class AMQPEndpoint {
+  //@todo fix jsdoc
+  constructor(connection, params = {}) {
+    this._connection = connection;
+    this._channel = null;
+    this._params = params;
+  }
+
+  async start() {
+    if (this._channel) {
```

Comment:
Theoretically, we have race condition here.
Example: we  call start. In other thread, before first thread finishes, we call start again.
Second call will call this._connection.createChannel()